project1/application.py

```diff
@@ -31,34 +32,51 @@
 
 @app.route("/")
 def index():
-    return render_template("register.html")
+    return render_template("register.html", flag=True)
 
 @app.route("/register",methods=["GET","POST"])
 def register():
     if (request.method == "POST"):
         username = request.form.get("uname")
         password = request.form.get("pswd")
         dt = datetime.datetime.now()
-        data = Data(Username=username,Password=password,Timestamp=dt)
-        db.session.add(data)
-        db.session.commit()
+        data2 = Data.query.all()
+        for user in data2:
+            if username == user.Username:
+                # return render_template("user.html")
+                return "<h2 Style='color: red;text-align:center'>You already have registered !Please Login </h2>"
         if not username:
             text = "Please enter username to register"
             return render_template("usernames.html", name=text, msg="ERROR")
         elif not password:
             text="Please provide password"
             return render_template("usernames.html", name=text ,msg="ERROR")
         else:
-            # text = "Success"
+            data = Data(Username=username,Password=password,Timestamp=dt)
+            db.session.add(data)
+            db.session.commit()
             return render_template("usernames.html",msg="SUCCESS")
-
-    return render_template("register.html")
+    return render_template("register.html", flag=True)
 
 @app.route("/admin")
 def admin():
     data1 = Data.query.all()
     return render_template("userlist.html", name=data1)
 
+@app.route("/auth",methods=["GET","POST"])
+def userhome():
+    if (request.method == "POST"):
+        username = request.form.get("uname")
+        password = request.form.get("pswd")
+        data3 = Data.query.all()
+        for user in data3:
+            if user.Username == username:
+                if user.Password == password:
+                    return render_template("user.html")
+        return render_template("register.html",flag=False)
+    if (request.method == "GET"):
+        return redirect(url_for('register'))
+
```

Comment:
You have to check for blank username and password here too. Please add those checks.

Reply:
done